man/cat1/xyzpoints.1



XYZPOINTS(1)             TIPSY COMMANDS              XYZPOINTS(1)



NAME
     xyzpoints - write the x,y, and z coordinates of  the  parti-
     cles to a file

SYNOPSIS
ALIASES
DESCRIPTION
SEE ALSO












































Katz and Quinn ReleasLeas2t.0change: 22 MARCH 1994                  1








code/ChangeLog
Wed Jan 11 13:09:36 1995  Tom Quinn  <trq@hermes.astro.washington.edu>

	* Makefile.in (SRCS): added alloca.c so it will be included in the
	distribution.

Wed Dec 28 13:42:47 1994  Tom Quinn  <trq@hermes.astro.washington.edu>

	* readmark.c: close file after it is read.

	* main.c: RedrawViewport(): make can_size the maximum of
	can_height and can_width.

	* canvas_resize_proc.c: Same as above.

Mon Dec  5 09:39:44 1994  Tom Quinn  <trq@hermes.astro.washington.edu>

	* xray_lum_load.c, star_lum.c, star_lum_redshift.c: Changed
	argument to my_gets() to blank string.

Fri Dec  2 08:21:50 1994  Tom Quinn  <trq@hermes.astro.washington.edu>

	* Makefile.in (READLINE_SRC): delete config.h.
	(tipsy): added dependence on ALLOCA.

	* fits.c: Added test for endianness.

Thu Dec  1 15:14:58 1994  Tom Quinn  <trq@hermes.astro.washington.edu>

	* vista.c: fixed initialization of quantity[].

	* find_shape.c: Forced inertia_cm to be symmetric.  This avoid
	problems with jacobi().

Tue Nov  1 11:26:56 1994  Tom Quinn  <trq@hermes.astro.washington.edu>

	* Makefile.in : fits_xray.c is now gone.  fits() is used instead.
	defs.h, xray.c, vista.c, fits.c and arguments.c are also changed.

Mon Oct 31 09:43:29 1994  Tom Quinn  <trq@hermes.astro.washington.edu>

	* main.c: RedrawColorbar(): fixed off-by-one drawing errors.

Wed Oct 12 14:27:16 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* Created version 2.1.2.

	* absorb.c: New function for calculating aborbtion spectra.

Tue Sep  6 16:06:02 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* readline/: Replaced with readline-2.0 from prep.ai.mit.edu.

	* my_gets.c: Changed cast to match new external from readline/.

Fri Jul 29 10:25:49 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* project.c: Ensure that the axes projections do not overflow.

Fri Jul 15 17:28:15 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* profile.c: Check to see if fopen() fails.

Wed Jul 13 09:46:18 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* defs.h: Decreased INTMAX to 2^29-1.  (One bit for sign, one bit
	to do INTMAX - (-INTMAX), and one bit for zoom out.

Tue Jul 12 15:55:17 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* canvas_resize_proc.c: Call plot_sub() and draw_label() for
	servers that automatically do backing store and don't generate an
	expose event.

	* main.c: ZoomIn(), ZoomOut(), Pan(), SetBox(): Ignore button
	presses in inactive windows.
	
	* main.c, window.c: Changed ResizeRequest to ConfigureNotify.

Sat Jun  4 13:12:06 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* fits.c, and fits_xray.c: Changed write() to fwrite() for
	portability.

Thu May  5 10:53:45 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* project.c: Increase clipping range of velocities to -2*INTMAX <
	v < 2*INTMAX.
	* defs.h: Change INTMAX to 2^30-1 to accommodate the above
	clipping.

Tue Apr 19 18:00:14 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* comm.h: NSK: added pcenter and writeboxrot commands.

Mon Apr 18 14:14:45 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* main.c: Add "unzoom" button.

Fri Apr 15 16:01:56 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* plot_all.c: Fixed bug with color of marked particles.
	Deleted extra conversion to screen coordinates.
	
	* maxpos.c: Make the margins a little smaller.

Thu Mar 31 14:03:11 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* plot_all.c: Changed XDrawArc to XFillArc for vector draws.

Fri Mar  4 18:09:25 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* Added Neal's changes for read_vector() and "master".

Mon Feb 28 11:51:39 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* openascii.c: Added code from Peter Teuben to uncompress files.

Thu Feb 24 13:21:50 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* Changed current_plot logical to current_project and
	current_color.

Tue Feb 22 11:24:05 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* readline: Updated to readline from gdb-1.12

Thu Feb 17 11:46:07 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* plot_all.c: Implimented a software "z-buffer".  Only draw
	particles with the highest particle color.

Wed Feb 16 13:21:35 1994  Tom Quinn  (trq@hermes.astro.washington.edu)

	* plot_all.c: Merged plot_all and plot_vel into one routine.  A
	new routine all_color() is created.  Changes were also needed in
	the view_*() routines.

Sat Nov 13 11:35:01 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* readmark.c: Fixed typo for increasing ndark.

Wed Nov  3 14:35:56 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* plot_all.c: Deleted skip of XClearWindow on first time through.

Thu Oct 28 10:23:50 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* hard.c: Added movie parameter.

Tue Oct 26 11:29:49 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* alloca.c: New file in case system doesn't provide alloca().

	* vista.c: Included <stdlib.h> for malloc() definition.

Mon Oct 25 19:16:52 1993  Tom Quinn  (trq@virgo)

	* main.c: Attempt to find a depth 8 PseudoColor visual.

Thu Oct 21 08:56:32 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* main.c: Set backgrounds of gc_color[] to black.

Tue Oct 19 10:56:36 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* defs.h, main.h, make_macro.c, read_macro.c, write_macro.c
	run_macro.c: re-vamped macro storage using linked lists.

Mon Oct 18 15:11:34 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* draworient.c: Added cast from Arg to long.

Fri Oct 15 10:04:49 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* readmark.c: Added a check for index < 1.

	* plot_sub.c: Added a check for binary_loaded.
	Added check for boxes_loaded[0].

Wed Oct  6 09:13:49 1993  Tom Quinn  (trq@hermes.astro.washington.edu)

	* zoom_window.c: Added zoom command.  Also changed comm.h and
	arguments.c.

Tue Oct  5 17:27:43 1993  Tom Quinn  (trq@hermes.astro.washington.edu

code/xyzpoints.c
#include "defs.h"
#include "fdefs.h"
void
xyzpoints(job)
    char job[MAXCOMM] ;
{
    char command[MAXCOMM] ;
    char file[MAXCOMM] ;
    char particle_type[MAXCOMM] ;
    struct gas_particle *gp, *lastgp ;
    struct dark_particle *dp, *lastdp ;
    struct star_particle *sp, *lastsp ;

    if (sscanf(job,"%s %s %s",command,particle_type,file) == 3) { 
	if (binary_loaded) {
	    lastgp = gas_particles + header.nsph ;
	    lastdp = dark_particles + header.ndark ;
	    lastsp = star_particles + header.nstar ;
	    sprintf(hardfile.name,"%s",file) ;
	    hardfile.ptr = fopen(hardfile.name,"w");
	    if(strcmp(particle_type,"gas")==0 ||
		    strcmp(particle_type,"all")==0){
		for(gp=gas_particles; gp < lastgp ; gp++) {
		    fprintf(hardfile.ptr,"%g %g %g\n",gp->pos[0],gp->pos[1],
			    gp->pos[2]) ;
		}
	    }
	    if(strcmp(particle_type,"dark")==0 ||
		    strcmp(particle_type,"all")==0){
		for(dp=dark_particles; dp < lastdp;  dp++){
		    fprintf(hardfile.ptr,"%g %g %g\n",dp->pos[0],dp->pos[1],
			    dp->pos[2]) ;
		}
	    }
	    if(strcmp(particle_type,"star")==0 ||
		    strcmp(particle_type,"all")==0){
		for(sp=star_particles; sp < lastsp; sp++){
		    fprintf(hardfile.ptr,"%g %g %g\n",sp->pos[0],sp->pos[1],
			    sp->pos[2]) ;
		}
	    }
	    fclose(hardfile.ptr) ;
	}
	else {
	    printf("<sorry, binary file is not loaded, %s>\n",title) ;
	}
    }
    else {
	input_error(command) ;
    }
}
